src/contracts/pods/EigenPod.sol

```diff
@@ -164,267 +175,355 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
+        * actual validator balance by 0.25 ETH.  In EigenLayer, we calculate our own "effective reskated balance" which is a further pessimistic
+        * view of the validator's effective balance.
+        */
+        uint64 validatorEffectiveBalanceGwei = Endian.fromLittleEndianUint64(validatorFields[BeaconChainProofs.VALIDATOR_BALANCE_INDEX]);
+
+        // verify ETH validator proof
+        bytes32 latestBlockHeaderRoot = eigenPodManager.getBeaconChainStateRoot(oracleTimestamp);
+
+        // verify that the provided state root is verified against the oracle-provided latest block header
+       BeaconChainProofs.verifyStateRootAgainstLatestBlockHeaderRoot(proofs.beaconStateRoot, latestBlockHeaderRoot, proofs.latestBlockHeaderProof);
+
+
+        BeaconChainProofs.verifyValidatorFields(
+            validatorIndex,
+            proofs.beaconStateRoot,
+            proofs.validatorFieldsProof,
+            validatorFields
+        );
+
+        // set the status to active
+       validatorInfo.status = VALIDATOR_STATUS.ACTIVE;
+
+        emit ValidatorRestaked(validatorIndex);
+
+        //record validator's new restaked balance
+        validatorInfo.restakedBalanceGwei = _calculateRestakedBalanceGwei(validatorEffectiveBalanceGwei);
+
+        //record validatorInfo update in storage
+        _validatorPubkeyHashToInfo[validatorPubkeyHash] = validatorInfo;
+
+        return validatorInfo.restakedBalanceGwei * GWEI_TO_WEI;
+    }
+
+    function _verifyAndProcessWithdrawal(
         BeaconChainProofs.WithdrawalProofs calldata withdrawalProofs, 
         bytes calldata validatorFieldsProof,
         bytes32[] calldata validatorFields,
         bytes32[] calldata withdrawalFields,
         uint256 beaconChainETHStrategyIndex,
-        uint64 oracleBlockNumber
+        uint64 oracleTimestamp
     )
-        external
-        onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_WITHDRAWAL)
-        onlyNotFrozen
+        internal
         /** 
-         * Check that the provided block number being proven against is after the `mostRecentWithdrawalBlockNumber`.
+         * Check that the provided block number being proven against is after the `mostRecentWithdrawalTimestamp`.
          * Without this check, there is an edge case where a user proves a past withdrawal for a validator whose funds they already withdrew,
          * as a way to "withdraw the same funds twice" without providing adequate proof.
-         * Note that this check is not made using the oracleBlockNumber as in the `verifyWithdrawalCredentials` proof; instead this proof
-         * proof is made for the block number of the withdrawal, which may be within 8192 slots of the oracleBlockNumber. 
+         * Note that this check is not made using the oracleTimestamp as in the `verifyWithdrawalCredentials` proof; instead this proof
+         * proof is made for the block number of the withdrawal, which may be within 8192 slots of the oracleTimestamp. 
```

Comment:
still talks about block numbers when it has been modified to use timestamps. also refers to a magic number (8192) rather than a variable name.